File: python/src/gestorAplicacion/ventas/Servicio.py
from gestorAplicacion.ventas.Iva import Iva

class Servicio(Iva):

    def __init__(self, nombre, precio = 0):
        from gestorAplicacion.general.Inventario import Inventario
        super().__init__()
        self._nombre = nombre
        self._precio = self.calcularPrecio(precio)    
        Inventario.agregarProducto(self)

    def getNombre(self):
        return self._nombre  
    def setNombre(self, nombre):
        self._nombre = nombre

    def getPrecio(self):
        return self._precio  
    def setPrecio(self, precio):
        self._precio = precio

    def calcularPrecio(self, precio):
        return round(precio*(1 + self.IVA))

    def consultarDisponibilidad(self, fechaSoliciud):
        from gestorAplicacion.general.Inventario import Inventario
        listaEmpleados = Inventario.getListadoEmpleados()
        empleadosDisponibles = []
        for empleado in listaEmpleados:
            if (empleado.consultarDisponibilidadEmpleado(self, fechaSoliciud)):
                empleadosDisponibles.append(empleado)
        return empleadosDisponibles


    def __str__(self):
        return self._nombre


   

File: src/gestorAplicacion/ventas/Servicio.java
package gestorAplicacion.ventas;

import java.time.LocalDate;
import java.util.ArrayList;

import gestorAplicacion.general.Inventario;
import gestorAplicacion.personas.Empleado;

import java.io.Serializable;
/**
 * Esta clase implementa la interface Iva y tiene como finalidad definir todos lo atributos y métodos de un servicio que
 * será adquirido por un cliente y prestado por un empleado
 * 
 * @author Mateo Alvarez Lebrum
 * @author Alejandro Alvarez Botero
 * @author Miguel Angel Barrera Bustamante
 * @author Alejandra Barrientos Grisales
 */

public class Servicio implements Iva, Serializable {
	private static final long serialVersionUID = 1L;

	private String nombre;
	private int precio;

	public Servicio(String nombre, int precio) {
		this.nombre = nombre;
		this.precio = calcularPrecio(precio);
		Inventario.agregarServicio(this);
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public int getPrecio() {
		return precio;
	}

	public void setPrecio(int precio) {
		this.precio = precio;
	}

	/**
	 * Calcula el precio teniendo en cuenta el IVA
	 * @param precio
	 * @return precio con Iva
	 */
	public int calcularPrecio(int precio) {
		return (int) Math.round(precio + precio * IVA);
	}

	/**
	 * Separa los empleados que estén disponibles para prestar un servicio en una fecha determinada
	 * @param fechaSolicitud
	 * @return empleadosDisponibles
	 */
	public ArrayList<Empleado> consultarDisponibilidad(LocalDate fechaSolicitud) {
		ArrayList<Empleado> listaEmpleados = Inventario.getListadoEmpleados();
		ArrayList<Empleado> empleadosDisponibles = new ArrayList<Empleado>();
		for (Empleado empleado : listaEmpleados) {
			if (empleado.consultarDisponibilidadEmpleado(this, fechaSolicitud)) {
				empleadosDisponibles.add(empleado);
			}
		}
		return empleadosDisponibles;

	}

	public String toString() {
		return nombre;
	}
}
